clang/lib/Sema/SemaCast.cpp

```diff
@@ -446,7 +446,12 @@ static bool tryDiagnoseOverloadedCast(Sema &S, CastType CT,
     : InitializationKind::CreateCast(/*type range?*/ range);
   InitializationSequence sequence(S, entity, initKind, src);
 
-  assert(sequence.Failed() && "initialization succeeded on second try?");
+  // It could happen that a constructor failed to be used because
+  // it requires a temporary of a broken type. Still, it will be found when
+  // looking for a match.
+  if (!sequence.Failed())
```

Comment:
Does this mean that `destType->containsErrors()` is true? Perhaps both should be true otherwise assert?

Reply:
I have checked, and the destination type has no errors. I guess because the type itself is ok, the problem is that the constructor from `const char*` failed to be instantiated.